earthfile2llb/withdockerrun.go

```diff
@@ -164,13 +176,51 @@ func (wdr *withDockerRun) Run(ctx context.Context, args []string, opt WithDocker
 		case <-ctx.Done():
 		}
 	}
-	// Sort the tar list, to make the operation consistent.
-	sort.Slice(wdr.tarLoads, func(i, j int) bool {
-		if wdr.tarLoads[i].imgName == wdr.tarLoads[j].imgName {
-			return wdr.tarLoads[i].platform.String() < wdr.tarLoads[j].platform.String()
+
+	var localRegImages []string
+	if wdr.enableImageRegistry {
+		// These "closer" functions are called to ensure that all ephemeral
+		// registry images are released and cleaned up. This must be called
+		// after the meat of the WITH DOCKER RUN command finishes.
+		defer func() {
+			for _, l := range wdr.localRegLoads {
+				l.closer()
+			}
+		}()
+		// Sort the local registry pull list, to make the operation consistent.
+		sort.Slice(wdr.localRegLoads, func(i, j int) bool {
+			return wdr.localRegLoads[i].imgName < wdr.localRegLoads[j].imgName
+		})
+		for _, l := range wdr.localRegLoads {
+			errChan := l.errChan
+			wdr.c.opt.ErrorGroup.Go(func() error {
```

Comment:
I'm using the preexisting `ErrorGroup` here since it's already handling output. These errors don't need to be acted on in this context since a failed build will force a failure for the overall command. But there may be a better way to handle and/or present this error.